Formula/m1ddc.rb

```diff
@@ -0,0 +1,34 @@
+class M1ddc < Formula
+  desc "Control external displays (USB-C/DisplayPort Alt Mode) using DDC/CI on M1 Macs"
+  homepage "https://github.com/waydabber/m1ddc"
+  url "https://github.com/waydabber/m1ddc/archive/175cdf191df15ff4cff1cebef8c07f1d9f49bbf8.tar.gz"
+  version "175cdf191df15ff4cff1cebef8c07f1d9f49bbf8"
+  sha256 "dbdeba76e24266d82552fafb3ba8142a0c63300173dfbc96ff30a68cbaed25b0"
+  license "MIT"
+
+  head "https://github.com/waydabber/m1ddc.git", branch: "main"
+
+  depends_on arch: :arm
+  depends_on macos: :monterey
+  depends_on :macos
+
+  def install
+    system "make"
+    bin.install "m1ddc"
+  end
+
+  def caveats
+    on_macos do
+      on_intel do
+        <<~EOS
+          m1ddc requires Apple Silicon, consider using ddcctl on Intel instead.
+        EOS
+      end
+    end
+  end
+
+  test do
+    # Ensure helptext is rendered
+    assert_includes shell_output("#{bin}/m1ddc help", 1), "Controls volume, luminance"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hey @SMillerDev, thanks for your time reviewing this-

https://github.com/Homebrew/homebrew-core/commit/9c1b7dda8a2117907149ab314a48eb3d98ea86b0 failed building on [tests (13-arm64-3868771519-1)](https://github.com/Homebrew/homebrew-core/actions/runs/3868771519/jobs/6594473212) with `Could not find a suitable external display.`

I shifted this in https://github.com/Homebrew/homebrew-core/commit/cc7ba502cf6f6a0e34014404965022194fa96cfb to only check the helptext, but I've added a specific check for the error message in https://github.com/Homebrew/homebrew-core/commit/ddaa73c636afa3a7224583d3b894a303633b98b7 based on your advice that seems to be passing CI now.